### ISSUE TITLE
Can't type square brackets and backtick after upgrading to JupyterLab 4.1.0

### ISSUE THREAD

**Issue by piotrp88** [jupyter/notebook#7243]
I'm running JupyterLab on a Ubuntu Server 22.04.3 LTS running Python 3.10.12.

After upgrading to the latest available version (4.1.0) I can't type open and closing square brackets and backtick ` inside code inputs.

I connect to JupyterLab from a macos client running macOS Catalina 10.15.7.

This is the behaviour I get from the following browsers:
* Firefox 122.0.1 (the latest available version) I can type only the closing square bracket `]`. Can't type the backtick.
* Google Chrome 121.0.6167.139 (the latest available version) I can't type both the opening `[` and closing square bracket `]`. Can't type the backtick.
* Safari 15.6.1: the same as Chrome

In all browsers the backtick `` ` ``, opening `[` and closing `]` brackets can be typed only if the typing cursor is positioned at the beginning of a row inside the input text area. After typing one of the three symbols, you can't type other ones. To type other backticks or square brackets you have to reposition the cursor at the beginning of the text inside the row. If the row starts with spaces or tabs, then the typing can start from position zero or from any position in the row prior to the first non-whitespace character.
This odd behaviour applies to empty text areas and to multiline text area filled with text.

Taking for example the `[` symbol and the following text:

```
   the first line starts with three spaces

 the third line starts with one space (the second line was empty)
fourth line starts without whitespaces
    the fifth line starts with four spaces
```

this is what you get if you place the cursors at the beginning or between each line's initial whitespace characters:

```
[   the first line starts with three spaces

[ the third line starts with one space (the second line was empty)
fourth line starts without whitespaces
    [the fifth line starts with four spaces
```

In the example above, you can't type any other `[`, `]`, `` ` `` after the `[` previously typed characters.

**Comment by jtpio** [comment]
Thanks @piotrp88.

Could it be related to this upstream issue in JupyterLab? https://github.com/jupyterlab/jupyterlab/issues/15744

**Comment by piotrp88** [comment]
> Thanks @piotrp88.
> 
> Could it be related to this upstream issue in JupyterLab? [jupyterlab/jupyterlab#15744](https://github.com/jupyterlab/jupyterlab/issues/15744)
> 
> Which may then be fixed by [jupyterlab/jupyterlab#15761](https://github.com/jupyterlab/jupyterlab/pull/15761)

Yes, this is exactly the same issues. I'm sorry if I opened a bug on the jupyter/notebook repo, I was confused about the correct one.
It's not clear to me if it's possilbe (and how) to temporarly solve this issue before waiting for the bug to be solved in the next release. May be I am missing something (I am new to Jupyter).
Thanks

**Comment by jtpio** [comment]
According to the discussion in https://github.com/jupyterlab/jupyterlab/issues/15744 we may need to wait for a new release, as it changes the default settings.

Closing in favor of https://github.com/jupyterlab/jupyterlab/issues/15744 since this is basically the same issue. Don't hesitate to re-open if needed, thanks!
